.github/workflows/tag_and_release.yml

```diff
@@ -0,0 +1,88 @@
+
+name: Add Version Tag
+
+permissions:
+  id-token: write
+  contents: write
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  release_commit_check:
+    runs-on: ubuntu-latest
+    outputs:
+        release_commit_check: ${{ steps.check_version_bump.outputs.release_commit_check }}
+        new_version: ${{ steps.check_version_bump.outputs.new_version }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Check For Version Bump
+        id: check_version_bump
+        run: |
+          MESSAGE="${{ github.event.head_commit.message }}"
+          BUMP_REGEX="^chore\([0-9]\.[0-9]\.[0-9]\): update pyproject version.*"
```

Comment:
We should consider adding a section in the `CONTRIBUTING.md` for how to deploy a release now that this commit pattern exists.